src/uct/cuda/base/cuda_iface.h

```diff
@@ -50,6 +50,24 @@
     })
 
 
+#define UCT_CUDADRV_CTX_ACTIVE(_state)                             \
+    do {                                                           \
+        CUcontext cur_ctx;                                         \
+        CUdevice dev;                                              \
+        unsigned flags;                                            \
+                                                                   \
+        _state = 0;                                                \
+        /* avoid active state check if no cuda activity */         \
+        if ((CUDA_SUCCESS == cuCtxGetCurrent(&cur_ctx))            \
+            && (NULL != cur_ctx)) {                                \
```

Comment:
should we check NULL first?

Reply:
Actually `cur_ctx` gets populated to NULL in the [`cuCtxGetCurrent`](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__CTX.html#group__CUDA__CTX_1g8f13165846b73750693640fb3e8380d0) so we need check after that.

Reply:
Ah! thought cur_ctx is input.